### PR TITLE
[Issue 4379] [Java Client] Build auth from class and params in PulsarClientImpl

### LINKED ISSUE #4379: ClientConfigurationData Authentication not set in PulsarClientImpl
**Describe the bug**
1. Setting Auth class and auth param does not build authentication instance in ClientConfigurationData.java
2. The lombok builders don't set default values set to attributes in class ClientConfigurationData.java. So PulsarClientImpl(conf) crashes.

### PR DESCRIPTION
**Pull request by shiv4289**
Fixes #4379 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications
In pull#4284 we added authParams and authPluginClassName to ClientConfigurationData. Though these were set the client instantiation did not build Authentication from these.
In the same PR @Builder from lombok was added to ease building the config instances. Turns out lombok builders set to 0/null fields that are not set explicitly. This means the client instantiation crashed if numListenerThreads was not set in builder always. 

*Describe the modifications you've done.*
Construct Auth when init client from config pojo.
Add @Builder.default to attributes in client config classes.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (yes)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: ( no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by shiv4289** [comment]
run integration tests

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java [hunk 1]
```diff
@@ -142,6 +145,13 @@ public PulsarClientImpl(ClientConfigurationData conf, EventLoopGroup eventLoopGr
         state.set(State.Open);
     }
 
+    private void setAuth(ClientConfigurationData conf) throws PulsarClientException {
+        if (StringUtils.isBlank(conf.getAuthPluginClassName()) || StringUtils.isBlank( conf.getAuthParams()))
```
Comment (sijie) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java]:
```suggestion
        if (StringUtils.isBlank(conf.getAuthPluginClassName()) || StringUtils.isBlank( conf.getAuthParams())) {
            return;
        }
```

**Comment by shiv4289** [comment]
run integration tests

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java [hunk 2]
```diff
@@ -46,33 +45,31 @@
...
-    private transient String authPluginClassName;
-    @JsonIgnore
-    private transient Map<String, String> authParams;
+    @Builder.Default private transient Authentication authentication = new AuthenticationDisabled();
+    private String authPluginClassName;
+    private String authParams;
 
-    private long operationTimeoutMs = 30000;
-    private long statsIntervalSeconds = 60;
+    @Builder.Default private long operationTimeoutMs = 30000;
```
Comment (merlimat) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java]: All these annotations seem to go a bit against the spirit of Lombok :) which is there to reduce the boiler-plate code.

Could we instead remove the `@Builder` and rely on the declared defaults (and just use `setX()` ) ?

Comment (shiv4289) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java]: Builder is just a convenience layer so of course yes. Requires changes in tests that use builder so I will come back with all the changes in  next 3-4 days @merlimat :-)

**Comment by merlimat** [comment]
@shiv4289 It seems there's one more test to fix: 

```
2019-06-02\T\14:11:47.477 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:testCompile (default-testCompile) on project pulsar-flink: Compilation failure: Compilation failure: 
2019-06-02\T\14:11:47.477 [ERROR] /home/jenkins/jenkins-slave/workspace/pulsar_precommit_cpp/pulsar-flink/src/test/java/org/apache/flink/batch/connectors/pulsar/PulsarOutputFormatTest.java:[111,69] cannot find symbol
2019-06-02\T\14:11:47.477 [ERROR]   symbol:   method builder()
2019-06-02\T\14:11:47.477 [ERROR]   location: class org.apache.pulsar.client.impl.conf.ClientConfigurationData
2019-06-02\T\14:11:47.477 [ERROR] /home/jenkins/jenkins-slave/workspace/pulsar_precommit_cpp/pulsar-flink/src/test/java/org/apache/flink/batch/connectors/pulsar/PulsarOutputFormatTest.java:[114,75] cannot find symbol
2019-06-02\T\14:11:47.477 [ERROR]   symbol:   method builder()
2019-06-02\T\14:11:47.477 [ERROR]   location: class org.apache.pulsar.client.impl.conf.ProducerConfigurationData
2019-06-02\T\14:11:47.477 [ERROR] -> [Help 1]

```

**Comment by shiv4289** [comment]
run cpp tests

**Comment by shiv4289** [comment]
run java8 tests

**Pull request merged**